libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/ContinueConversationLater.cs

```diff
@@ -32,17 +32,21 @@ public class ContinueConversationLater : Dialog
         /// The Kind name for this dialog.
         /// </summary>
         [JsonProperty("$kind")]
-        public const string Kind = "AzureQueues.ContinueConversationLater";
+        public const string Kind = "Microsoft.ContinueConversationLater";
+
+        private readonly QueueStorage _queueStorage;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ContinueConversationLater"/> class.
         /// </summary>
+        /// <param name="queueStorage">The queue to store the Activity to continue the conversation.</param>
         /// <param name="callerPath">The full path of the source file that contains this called.</param>
         /// <param name="callerLine">The line within the source file that contains this caller.</param>
         [JsonConstructor]
```

Comment:
I think the JsonConstructor needs to be 
`public ContinueConversationLater([CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)`

This class should have a load method retrieving the QueueStorage from TurnState, similar to how BeginSkill loads the DialogOptions:

https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/BeginSkill.cs#L313

this.dialogManager.InitialTurnState.Set(queuStorage);

Reply:
or take the config parameters, and create the QueueStorage when needed?  (but, TurnState approach seems to be the pre-existing pattern)

Reply:
I am removing QueueStorage from the constructor and getting it from TurnState.